vm/ubpf_jit_x86_64.c

```diff
@@ -51,18 +51,24 @@ static int platform_nonvolatile_registers[] = {
 static int platform_parameter_registers[] = {
     RCX, RDX, R8, R9
 };
-#define RCX_ALT R15
+#define RCX_ALT R10
+// Register assignments:
+// BPF R0-R4 are "volatile"
+// BPF R5-R10 are "non-volatile"
+// Map BPF volatile registers to x64 volatile and map BPF non-volatile to
+// x64 non-volatile.
+// Avoid R12 as we don't support encoding modrm modifier for using R12.
 static int register_map[REGISTER_MAP_SIZE] = {
     RAX,
     R15,
     RDX,
     R8,
     R9,
-    R10,
-    R11,
-    R12,
-    R13,
     R14,
+    R15,
```

Comment:
R15 is repeated.

Reply:
Also, https://en.wikipedia.org/wiki/X86_calling_conventions#Microsoft_x64_calling_convention mentions that R15 is non-volatile, so it shouldn't be mapped to a volatile ebpf register, right?

Reply:
Good catch. That was supposed to be R10, not R15. I fixed the RCX_ALT to point to R10 but didn't fix it here.